### PR TITLE
feat(console): connector config

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

Add config area in connector details page.

JSON Editor will be implemented in LOG-1708

`{saveError && <div>{saveError}</div>}` will be refactored once the design is finished.

## Linear Issue Reference

LOG-1722

## Testing

image
image

### CODE REVIEW

File: packages/console/src/pages/ConnectorDetails/index.tsx [hunk 1]
```diff
@@ -28,6 +34,44 @@ const ConnectorDetails = () => {
...
+        .json<ConnectorDTO>();
+      // TODO: show a toast, awaiting PR #354.
+    } catch (error: unknown) {
+      if (error instanceof SyntaxError) {
+        setSaveError(t('connector_details.save_error_json_parse_error'));
+      } else if (error instanceof HTTPError) {
+        const { message } = (await error.response.json()) as RequestErrorBody;
+        setSaveError(message);
+      } else {
+        console.error(error);
```
Comment (gao-sun) [inline packages/console/src/pages/ConnectorDetails/index.tsx:68]: we need to throw and handle general errors, maybe a `<ErrorBoundary>` component to show a toast?

can do it later, let's add an issue for it

File: packages/console/src/pages/ConnectorDetails/index.tsx [hunk 2]
```diff
@@ -28,6 +34,44 @@ const ConnectorDetails = () => {
...
+      return;
+    }
+
+    try {
+      const configJson = JSON.parse(config) as JSON;
+      setIsSubmitLoading(true);
+      await ky
+        .patch(`/api/connectors/${connectorId}`, { json: { config: configJson } })
+        .json<ConnectorDTO>();
+      // TODO: show a toast, awaiting PR #354.
```
Comment (gao-sun) [inline packages/console/src/pages/ConnectorDetails/index.tsx:60]: 354 merged

**Review by gao-sun — APPROVED** [review]

File: packages/core/src/middleware/koa-connector-error-handle.ts [hunk 3]
```diff
@@ -22,7 +22,7 @@ export default function koaConnectorErrorHandler<StateT, ContextT>(): Middleware
           throw new RequestError(
             {
               code: 'connector.invalid_config',
-              status: 500,
+              status: 400,
```
Comment (demonzoo) [inline packages/core/src/middleware/koa-connector-error-handle.ts:25]: Why 400? Invalid config should be an internal error, thus 500?
400 indicates a bad request

Comment (xiaoyijun) [reply packages/core/src/middleware/koa-connector-error-handle.ts:25]: `400` represents a user error.

Comment (wangsijie) [reply packages/core/src/middleware/koa-connector-error-handle.ts:25]: is the user's fault, user pass in an invalid config, throw 400 and reject to save.

File: packages/console/src/pages/ConnectorDetails/index.tsx (hunk 1 above)
Comment (wangsijie) [reply packages/console/src/pages/ConnectorDetails/index.tsx:68]: LOG-1868

**Review by xiaoyijun — APPROVED** [review]
LGTM

File: packages/console/src/pages/ConnectorDetails/index.tsx (hunk 2 above)
Comment (wangsijie) [reply packages/console/src/pages/ConnectorDetails/index.tsx]: updated

File: packages/core/src/middleware/koa-connector-error-handle.ts (hunk 3 above)
Comment (demonzoo) [reply packages/core/src/middleware/koa-connector-error-handle.ts:25]: image

After digging into this, I think 400 serves our purpose. So I'm OK with 400 now

**Pull request merged**